rules/CodingStyle/Rector/Assign/NestedTernaryToMatchTrueRector.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\Assign;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\Assign;
+use PhpParser\Node\Expr\Match_;
+use PhpParser\Node\Expr\Ternary;
+use PhpParser\Node\MatchArm;
+use Rector\CodingStyle\ValueObject\ConditionAndResult;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\CodingStyle\Rector\Assign\NestedTernaryToMatchTrueRector\NestedTernaryToMatchTrueRectorTest
+ */
+final class NestedTernaryToMatchTrueRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Convert nested ternary expressions to match(true) statements', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function getValue($input)
+    {
+        return $input === 1 ? 'one' : ($input === 2 ? 'two' : 'other');
+    }
+}
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function getValue($input)
+    {
+        return match (true) {
+            $input === 1 => 'one',
+            $input === 2 => 'two',
+            default => 'other',
+        };
```

Comment:
This specific case should match the `$input` value:

```suggestion
        return match ($input) {
            1 => 'one',
            2 => 'two',
            default => 'other',
        };
```

Reply:
Good point :+1:  I've added a fixture to cover this.